returnn/hdf.py

```diff
@@ -346,6 +366,278 @@ def run(self):
+        :param splits: List of segment files that list the segments per HDF.
+            The job creates one HDF per split.
+        :param concurrent: split up the list of splits into this many concurrent jobs.
+            Recommended is about one unit of concurrency per 1000h of audio.
+            Within job concurrency is handled by the multiprocessing library.
+        :param compress_format: optional compression format for the audio data.
+        :param compress_factor: compression factor for the audio data.
+        :param multi_channel_strategy: defines what should happen to multi-channel audio files.
+            Currently implemented are:
+            Mixdown(): Mix down all channels to one channel, default.
+            PickNth(n): Takes audio from n-th channel.
+        :param output_dtype: dtype that should be written in the hdf (supports float64, float32, int32, int16).
+            If writing compressed data, must be set to None as the compressed audio is always written as uint8 (raw bytes).
+        :param returnn_root: RETURNN repository
+        :param rounding: defines how timestamps should be rounded if they do not exactly fall onto a sample:
+            start_and_duration will round down the start time and the duration of the segment
+            rasr_compatible will round up the start time and round down the end time
+        :param round_factor: do the rounding based on a sampling rate that is scaled down by this factor
+        :param target_sampling_rate: desired sampling rate for the HDF, data will be resampled to this rate if needed
+        :param skip_on_error: If True, skip segments that fail to be processed (for known reasons only) instead
+            of failing the job.
+        """
+        self.bliss_corpus = bliss_corpus
+        self.splits = splits
+        assert concurrent > 0
+        self.concurrent = concurrent
+        if compress_format is not None:
+            assert compress_factor is not None
+            assert 0 <= compress_factor
+            assert output_dtype is None
+            self.output_dtype = "int32"
+        else:
+            assert output_dtype in ["float64", "float32", "int32", "int16"]
+            self.output_dtype = output_dtype
+        self.compress_format = compress_format
+        self.compress_factor = compress_factor
+        self.multi_channel_strategy = multi_channel_strategy or BlissToPcmHDFJob.Mixdown()
+        assert isinstance(self.multi_channel_strategy, (BlissToPcmHDFJob.Mixdown, BlissToPcmHDFJob.PickNth))
+        self.rounding = rounding
+        assert round_factor > 0
+        self.round_factor = round_factor
+        self.returnn_root = returnn_root
+        self.target_sampling_rate = target_sampling_rate
+        self.skip_on_error = skip_on_error
+
+        self.out_hdfs = [self.output_path(f"{i + 1:0d}.hdf") for i in range(len(splits))]
+
+        num_cores = min(len(splits) + 1, 16)
```

Comment:
Why does the `num_cores` scale with the `len(splits)`?
It seems to me that if at all it should rather depend on `len(splits)/concurrent` but then making that ratio larger would just make the job take more time ~and we should instead always just use as many cores as ffmpeg can comfortably handle.~ 
EDIT: I saw below this is used by multiprocessing in the python process and have a separate thread for that now.